src/Finder.php

```diff
@@ -33,4 +33,14 @@ public function __construct()
             ->exclude('vendor')
         ;
     }
+
+    /**
+     * Check if this finder wasn't extended.
+     *
+     * return bool
+     */
+    public function isOriginal()
+    {
+        return serialize(new self()) === serialize($this);
```

Comment:
nope
when you will change a property like filtering, yet still not configuring source where to find files, this will return false, wihch would be false positive.

Reply:
here is how we detect that finder from config file is fully defined:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.5/src/Console/ConfigurationResolver.php#L777

Reply:
When I will change a property like filtering I will change finder, so false would be correct.

I'm not aiming to check if finder is fully defined, but to check if maybe someone extended Finder like [here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.5/.php_cs.dist#L50).

I think we are not interested if Finder is fully defined, only if default one is overridden.

I'm open to suggestion how solve this differently as this is nicest solution I have found.

Reply:
if one has a not fully defined finder in his config and provided a path, we shall not display a warning, as he is not overwriting the finder.

only fully-defined finder could be overwritten with cli `path` argument.
if finder in config is not fully defined, cli `path` argument will fulfill it: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.5/src/Console/ConfigurationResolver.php#L807